docs/pages/guides/react-compiler.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: React Compiler
+description: Learn how to enable and use the React Compiler in Expo apps.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+
+> Warning: React Compiler is still experimental!
+
+The new [React Compiler](https://react.dev/learn/react-compiler) automatically memoizes components and hooks to enable fine-grained reactivity. This can lead to significant performance improvements in your app. The React Compiler is currently experimental and is not enabled by default. You can enable it in your app by following the instructions below.
+
+## Enabling React Compiler
+
+First, [check how compatible](https://react.dev/learn/react-compiler#checking-compatibility) your project is with the React Compiler.
+
+<Terminal cmd={['$ npx react-compiler-healthcheck@latest']} />
+
+This will generally verify if your app is following the [**rules of React**](https://react.dev/reference/rules).
+
+Next, toggle on the React Compiler experiment in your `app.json` file:
+
+```json app.json
+{
+  "experiments": {
+    "reactCompiler": true
+  }
+}
+```
+
+### Enabling the linter
+
+Additionally, you should use the ESLint plugin to continuously enforce the rules of React in your project.
+
+Do this by running `npx expo lint` to ensure Lint is setup in your app, then install the ESLint plugin:
+
+<Terminal cmd={['$ npx expo install eslint-plugin-react-compiler']} />
+
+Then update your ESLint configuration to include the plugin:
+
+```js .eslintrc.js
+// https://docs.expo.dev/guides/using-eslint/
+module.exports = {
+  extends: 'expo',
+  plugins: ['eslint-plugin-react-compiler'],
+  rules: {
+    'react-compiler/react-compiler': 'error',
+  },
+};
+```
+
+> In the future, all of these steps will be automated by Expo CLI.
```

Comment:
````suggestion
> In the future, all of these steps below will be automated by Expo CLI.

Additionally, you should use the ESLint plugin to continuously enforce the rules of React in your project.

<Step label="1">

Run [`npx expo lint`](/guides/using-eslint/#eslint) to ensure ESLint is setup in your app, then install the ESLint plugin for React Compiler:

<Terminal cmd={['$ npx expo install eslint-plugin-react-compiler']} />

</Step>

<Step label="2">

Update your ESLint configuration to include the plugin:

```js .eslintrc.js
// https://docs.expo.dev/guides/using-eslint/
module.exports = {
  extends: 'expo',
  plugins: ['eslint-plugin-react-compiler'],
  rules: {
    'react-compiler/react-compiler': 'error',
  },
};
```

</Step>
````

- Update the callout verbiage to indicate the steps being referenced are from this section.
- Move callout at the top of this section.
- Use `Step` to describe the procedure
- Add link to ESLint guide
- Mention in the first step that the command below installs ESLint plugin for React Compiler.